pytest_fixtures/component/discovery.py

```diff
@@ -29,3 +29,14 @@ def discovery_location(module_location, module_target_sat):
     discovery_loc = module_target_sat.update_setting('discovery_location', module_location.name)
     yield module_location
     module_target_sat.update_setting('discovery_location', discovery_loc)
+
+
+@pytest.fixture(scope='module')
+def provisioning_env(module_target_sat, discovery_org, discovery_location):
+    # Build PXE default template to get default PXE file
+    module_target_sat.cli.ProvisioningTemplate().build_pxe_default()
+    return module_target_sat.api_factory.configure_provisioning(
+        org=discovery_org,
+        loc=discovery_location,
+        os='Redhat {}'.format(module_target_sat.cli_factory.RHELRepository().repo_data['version']),
+    )
```

Comment:
Does this configure_provisioning works for you? also tests using this are being removed so do you still want to keep it?
```suggestion
    return module_target_sat.api_factory.configure_provisioning(
        org=discovery_org,
        loc=discovery_location,
        os=f'Redhat {module_target_sat.cli_factory.RHELRepository().repo_data['version'])}',
    )
```

Reply:
Didn't get the first part of the question about why it wouldn't work? , and do we have an alternative for this as you are saying it is being removed?

Reply:
We don't have any tests using this fixture/helper, so does it make sense to remove this?
And, configure_provisioning contains some old provisioning setup code and it might require some refactor, that's the reason I asked if it does work for you

Reply:
Yeah, it works fine!

Reply:
tests using the fixture provisioning_env are removed in this PR, but @adarshdubey-star wants to keep it for the future :)